pkg/datagatherer/k8s/generic.go

```diff
@@ -110,9 +112,41 @@ func (g *DataGatherer) Fetch() (interface{}, error) {
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+	// Redact Secret data
+	err = redactList(list)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
 	return list, nil
 }
 
+func redactList(list *unstructured.UnstructuredList) error {
+	// In principal we could only redact the list if it's kind is SecretList or
+	// a generic mixed List, however the test suite does not set the list kind
+	// and it is safer to always check for Secrets.
+	// Iterate over the items in the list.
+	for i := range list.Items {
+		// Determine the kind of items in case this is a generic 'mixed' list.
+		gvks, _, err := scheme.Scheme.ObjectKinds(list.Items[i].DeepCopyObject())
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		redactItem := false
+		for _, gvk := range gvks {
+			// If this item is a Secret then we need to redact it.
+			if gvk.Kind == "Secret" {
+				redactItem = true
+				break
+			}
+		}
+		// Redcat the Secret by overwriting its data.
+		if redactItem {
+			list.Items[i].Object["data"] = map[string]interface{}{}
```

Comment:
Hm - there's a few different ways this could be redacted...

1) Removing the `data` field altogether
2) Making `data` be `nil`
3) Setting `data` to an empty map (what currently happens here now).

I think (3) _does_ make sense to do, but thought I'd call out the various options here as there _are_ options :)